tests/ixia/pfcwd/files/pfcwd_basic_helper.py

```diff
@@ -0,0 +1,301 @@
+import time
+import dpkt
+from math import ceil
+
+from tests.common.helpers.assertions import pytest_assert
+from tests.common.fixtures.conn_graph_facts import conn_graph_facts,\
+    fanout_graph_facts
+from tests.common.ixia.ixia_fixtures import ixia_api_serv_ip, ixia_api_serv_port,\
+    ixia_api_serv_user, ixia_api_serv_passwd, ixia_api
+from tests.common.ixia.ixia_helpers import get_dut_port_id
+from tests.common.ixia.common_helpers import pfc_class_enable_vector,\
+    start_pfcwd_default, get_pfcwd_poll_interval, get_pfcwd_detect_time,\
+    get_pfcwd_restore_time
+
+from abstract_open_traffic_generator.capture import CustomFilter, Capture,\
+    BasicFilter
+from abstract_open_traffic_generator.flow import DeviceTxRx, TxRx, Flow, Header,\
+    Size, Rate,Duration, FixedSeconds, FixedPackets, PortTxRx, PfcPause
+from abstract_open_traffic_generator.flow_ipv4 import Priority, Dscp
+from abstract_open_traffic_generator.flow import Pattern as FieldPattern
+from abstract_open_traffic_generator.flow import Ipv4 as Ipv4Header
+from abstract_open_traffic_generator.flow import Ethernet as EthernetHeader
+from abstract_open_traffic_generator.control import State, ConfigState,\
+    FlowTransmitState, PortCaptureState
+from abstract_open_traffic_generator.result import FlowRequest, CaptureRequest
+
+PAUSE_FLOW_NAME = "Pause Storm"
+DATA_FLOW1_NAME = "Data Flow 1"
+DATA_FLOW2_NAME = "Data Flow 2"
+DATA_PKT_SIZE = 1024
+IXIA_POLL_DELAY_SEC = 2
+DEVIATION = 0.2
+
+def run_pfcwd_basic_test(api,
+                         testbed_config,
+                         conn_data,
+                         fanout_data,
+                         duthost,
+                         dut_port,
+                         prio_list,
+                         prio_dscp_map,
+                         trigger_pfcwd):
+
+    start_pfcwd_default(duthost)
+
+    """ Get the ID of the port to test """
+    port_id = get_dut_port_id(dut_hostname=duthost.hostname,
+                              dut_port=dut_port,
+                              conn_data=conn_data,
+                              fanout_data=fanout_data)
+
+    pytest_assert(port_id is not None,
+                  'Fail to get ID for port {}'.format(dut_port))
+
+    poll_interval_sec = get_pfcwd_poll_interval(duthost) / 1000.0
+    detect_time_sec = get_pfcwd_detect_time(host_ans=duthost, intf=dut_port) / 1000.0
+    restore_time_sec = get_pfcwd_restore_time(host_ans=duthost, intf=dut_port) / 1000.0
+
+    if trigger_pfcwd:
```

Comment:
Can you point to the README here so it is easier to understand the calculations done below?